packages/@aws-cdk/pipelines/lib/pipeline.ts

```diff
@@ -441,6 +442,9 @@ class AssetPublishing extends CoreConstruct {
       this.generateAssetRole(command.assetType);
     }
 
+    this.assetPublishingRoles[command.assetType] = this.assetPublishingRoles[command.assetType] ?? new Set();
+    this.assetPublishingRoles[command.assetType].add(command.assetPublishingRoleArn);
```

Comment:
would this be too weird?
```suggestion
    (this.assetPublishingRoles[command.assetType] = this.assetPublishingRoles[command.assetType] ?? new Set()).add(command.assetPublishingRoleArn);
```

Reply:
Your version doesn't quite do what mine does; if the Set already exists, it'll be a no-op, rather than adding the ARN to the set. I think the below works, but it was just a bit lengthy for my tastes. I don't feel terribly strongly about it though.

```suggestion
    this.assetPublishingRoles[command.assetType] = (this.assetPublishingRoles[command.assetType] ?? new Set()).add(command.assetPublishingRoleArn);
```